src/coreclr/jit/instr.cpp

```diff
@@ -2024,31 +2024,38 @@ instruction CodeGenInterface::ins_Store(var_types dstType, bool aligned /*=false
 // Return Value:
 //   the instruction to use
 //
-// Notes:
-//   The function currently does not expect float srcReg with integral dstType and will assert on such cases.
-//
 instruction CodeGenInterface::ins_StoreFromSrc(regNumber srcReg, var_types dstType, bool aligned /*=false*/)
```

Comment:
it looks like it is another function that we should use always instead of `ins_Store`  but currently we don't.
I have updated arm64 to match arm32 and left xarch as it was for now.